Prose/ProseLib/Sources/MainWindowFeature/MainScreen.swift

```diff
@@ -40,92 +54,91 @@ public struct MainScreen: View {
 
 // MARK: Reducer
 
-public let mainWindowReducer: Reducer<
+public let mainWindowReducer = Reducer<
     MainScreenState,
     MainScreenAction,
     MainScreenEnvironment
-> = sidebarReducer._pullback(
-    state: \MainScreenState.sidebar,
-    action: CasePath(MainScreenAction.sidebar),
-    environment: \MainScreenEnvironment.sidebar
-)
+>.combine([
+    sidebarReducer.pullback(
+        state: \.sidebar,
+        action: CasePath(MainScreenAction.sidebar),
+        environment: { _ in }
+    ),
+    unreadReducer._pullback(
+        state: (\MainScreenState.route).case(CasePath(MainScreenState.Route.unreadStack)),
+        action: CasePath(MainScreenAction.unreadStack),
+        environment: { _ in .stub }
+    ),
+    conversationReducer._pullback(
+        state: (\MainScreenState.route).case(CasePath(MainScreenState.Route.chat)),
+        action: CasePath(MainScreenAction.chat),
+        environment: { _ in .stub }
+    ),
+    Reducer { state, action, _ in
+        switch action {
+        case .sidebar(.selection(.unreadStack)):
+            state.route = .unreadStack(.init())
```

Comment:
So we're resetting the `UnreadStackState` every time the user leaves and comes back? Isn't this a bad UX?

I had cached it at some point, what do you suggest?

Reply:
Unread messages are not static. You don't have 3 unread messages between viewing the "Unread Stack". You receive new messages which are unread by default, so just caching the state isn't a solution.

Reply:
Ok, fair enough